src/gui/dialogs/file_dialog.hpp

```diff
@@ -0,0 +1,288 @@
+	 */
+	tfile_dialog& set_message(const std::string& value)
+	{
+		msg_ = value;
+		return *this;
+	}
+
+	/**
+	 * Gets the current file selection.
+	 *
+	 * @returns An absolute path to the selected file.
+	 */
+	std::string path() const;
+
+	/**
+	 * Sets the initial file selection.
+	 *
+	 * If the path is found to refer to a file (more specifically, any
+	 * non-directory object), that file is initially selected on the directory
+	 * contents view and the file name box is set to contain its name. If the file
+	 * does not exist, but the path leading up to it does, the directory contents
+	 * view displays that path and there isn't an initial file selection or name
+	 * set unless set_filename() is used first.
+	 *
+	 * If you want to set an initial file name hint/template, use set_filename()
+	 * <b>after</b> calling this method.
+	 */
+	tfile_dialog& set_path(const std::string& value);
+
+	/**
+	 * Sets the initial file name input but not the path.
+	 *
+	 * The file name needs not exist in the initial path selected with set_path().
+	 *
+	 * If this is used before set_path() and the path passed there points to a
+	 * file, that file name will replace the one given here.
+	 */
+	tfile_dialog& set_filename(const std::string& value);
+
+	/**
+	 * Sets the default file extension for file names in save mode.
+	 *
+	 * When this is set to a non-empty string and save mode is active, selecting
+	 * file entries will cause their name portions to be highlighted in the name
+	 * text box if their extensions match the provided template, and any time the
+	 * text box is cleared it will position the cursor before the extension as a
+	 * hint for the user.
+	 *
+	 * The value provided to this method should be preceded by a dot if
+	 * applicable (e.g. ".cfg").
+	 */
+	tfile_dialog& set_extension(const std::string& value)
+	{
+		extension_ = value;
+		return *this;
+	}
+
+	/**
+	 * Whether user interface elements for manipulating existing objects are provided.
+	 */
+	bool read_only() const
+	{
+		return read_only_;
+	}
+
+	/**
+	 * Whether to provide user interface elements for manipulating existing objects.
+	 *
+	 * This is initially disabled.
+	 */
+	tfile_dialog& set_read_only(bool value)
+	{
+		read_only_ = value;
+		return *this;
+	}
+
+	/**
+	 * Returns whether save mode is enabled.
+	 *
+	 * See set_save_mode() for more information.
+	 */
+	bool save_mode() const
+	{
+		return save_mode_;
+	}
+
+	/**
+	 * Sets the dialog's behavior on non-existent file name inputs.
+	 *
+	 * This is initially disabled.
+	 *
+	 * When save mode is enabled, file names entered into the dialog by the user
+	 * need not exist already (but their parent directories still do). Otherwise,
+	 * the user is only able to select existing files.
+	 */
+	tfile_dialog& set_save_mode(bool value)
+	{
+		save_mode_ = value;
+		return *this;
+	}
+
+	/**
+	 * Sets the OK button label.
+	 *
+	 * By default, "Save" is used when save_mode is enabled, and "Open" otherwise.
```

Comment:
I forgot that this was a thing when wiring the dialog into the GUI1 filechooser callers.